worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java

```diff
@@ -222,12 +264,19 @@ private void loadEobRequests(ContractData contractData) throws InterruptedExcept
      * @param contractData object containing list of in progress requests
      * @param result       the page of beneficiaries that need requests to be created for them
      */
-    private void loadRequestBatch(ContractData contractData, CoveragePagingResult result) {
-
-        for (CoverageSummary summary : result.getCoverageSummaries()) {
-            Future<EobSearchResult> requestFuture = queuePatientClaimsRequest(summary, contractData);
-
-            contractData.addEobRequestHandle(requestFuture);
+    private void loadRequestBatch(ContractData contractData, CoveragePagingResult result, int searchBatchSize) {
+        Queue<CoverageSummary> coverageSummaries = new LinkedList<>(result.getCoverageSummaries());
+        int actualBatchSize = searchBatchSize == 0 ? 1 : searchBatchSize;
+
+        while (coverageSummaries.size() > 0) {
+            List<CoverageSummary> subList = new ArrayList<>(actualBatchSize);
+            for (int i = 0; i < actualBatchSize; i++) {
```

Comment:
Suggest using idx instead of i for loop variables, much easer to search for.

One letter variable names are less desireable.

Reply:
Just wondering if you can use Java 8 here to split everything into batches instead of a for loop. Might be cleaner.

Reply:
I do not like idx. I will stubbornly hold on to the one place I consider single letter names standard and reasonable. I will also use j & k for any nested loops. 

I tried a few Java 8 versions and maybe there is a way, but using a linked list, removing items as I went and processing it as I parse the list I found to be much more memory efficient and it ran faster.